database/store.py

```diff
@@ -26,45 +27,196 @@ def applications_index(self) -> dict:
+            account_id, fund_id, round_id, sections
         )
         self._applications.update({application_id: new_application})
         return new_application
 
+    def _get_sections(self, fund_id: str, round_id: str):
+        fund = get_fund(fund_id)
+        fund_round = get_round(fund_id, round_id)
+        if fund and fund_round:
+            sections = fund_round_sections.get(":".join([fund_id, round_id]))
+            if not sections:
+                raise Exception(
+                    f"Could not find form sections for {fund_id} - {round_id}"
+                )
+            return sections
+        raise Exception(
+            f"Could not find fund round for {fund_id} - {round_id}"
+        )
+
+    def submit_application(self, application_id):
+        self._applications[application_id][
+            "date_submitted"
+        ] = datetime.datetime.now(datetime.timezone.utc).strftime(
+            "%Y-%m-%d %H:%M:%S"
+        )
+        self._update_statuses(application_id)
+        return self._applications[application_id]
+
+    def get_section(self, application_id, section_name):
+        return self._applications[application_id]["sections"][section_name]
+
+    def _find_value_by_key(self, data: dict, target):
+        for key, value in data.items():
+            if isinstance(value, dict):
+                yield from self._find_value_by_key(value, target)
+            elif key == target:
+                yield value
+
+    def put_section(self, application_id, section_name, new_json):
+        """
+        Updates a section of the application at section_name
+
+        Args:
+            application_id: The id of the application to update
+            section_name: The name of the section to update
+            new_json: The payload of data to update.
+                This currently expects a json object in the form:
+                {
+                    "questions": section_questions,
+                    "section_name": section_name,
+                    "metadata": section_metadata
+                }
+        """
+        # Find matching section, update with put data
+        section_index = None
+        for index, section in enumerate(
+            self._applications[application_id]["sections"]
+        ):
+            if section["section_name"].lower() == section_name.lower():
+                section_index = index
+                self._applications[application_id]["sections"][
+                    section_index
+                ] = new_json
+                # Update application statuses
+                self._update_statuses(application_id)
+        # Set last edited
+        self._applications[application_id][
+            "last_edited"
+        ] = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        # IF section includes "your-project-name" field
+        # THEN update application project name
+        project_name = list(
+            self._find_value_by_key(new_json, "your-project-name")
```

Comment:
⛏️ 🤔 Doesn't the use of list make the yield redundant?

Reply:
yes